crates/shared/src/order_validation.rs

```diff
@@ -380,7 +387,7 @@ impl OrderValidating for OrderValidator {
             from: owner,
             app_data: order.data.app_data,
         };
-        let quote = if !liquidity_owner {
+        let quote = if !liquidity_owner && order.data.fee_amount > U256::zero() {
```

Comment:
nit: Since the `if` condition changed slightly it would make sense to also update the comment in the `else` branch to mention limit orders as well.